src/components/search/FacetList.jsx

```diff
@@ -0,0 +1,109 @@
+import React, {
+  useCallback, useContext, useEffect, useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import { Collapsible, Input } from '@edx/paragon';
+import { connectRefinementList } from 'react-instantsearch-dom';
+import classNames from 'classnames';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faChevronUp, faChevronDown } from '@fortawesome/free-solid-svg-icons';
+import OutsideClickHandler from 'react-outside-click-handler';
+import { AppContext } from '@edx/frontend-platform/react';
+
+import './styles/FacetList.scss';
+
+const BaseFacetList = ({
+  title,
+  attribute,
+  items,
+  currentRefinement,
+  refine,
+}) => {
+  const { enterpriseConfig: { branding } } = useContext(AppContext);
+  const [isOpen, setIsOpen] = useState(false);
+
+  useEffect(() => {
+    function checkKeyAndCloseIfEsc({ key }) {
+      if (key === 'Escape') {
+        setIsOpen(false);
+      }
+    }
+    document.addEventListener('keydown', checkKeyAndCloseIfEsc);
+    return () => { // on unmount, remove event listener
+      document.removeEventListener('keydown', checkKeyAndCloseIfEsc);
+    };
+  }, []);
+
+  const handleOutsideClick = () => {
+    if (isOpen) {
+      setIsOpen(false);
+    }
+  };
+
+  const renderItems = useCallback(
+    () => {
+      if (!items.length) {
+        return (
+          <p>No options found.</p>
+        );
+      }
+      return (
+        <ul className="list-group">
+          {items.map((item, index) => (
+            <li key={item.label} className="list-group-item border-0">
+              {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+              <label htmlFor={`${attribute}-${index}`}>
+                <Input
+                  type="checkbox"
+                  id={`${attribute}-${index}`}
+                  checked={item.isRefined}
+                  onChange={() => refine(item.value)}
+                />
+                <span className={classNames('facet-item-label', 'ml-1', { 'is-refined': item.isRefined })}>
+                  {item.label}
+                </span>
+                <span className="badge badge-pill ml-2 bg-brand-primary text-brand-primary">
+                  {item.count}
+                </span>
+              </label>
+            </li>
```

Comment:
Nit: maybe a candidate for a helper function like `facetItem(item, index)`?

Reply:
Considering this JSX is already rendered through a `.map`, it already is sort of in a helper function? 😉 

I was going to say we could create a new component for this JSX and render the component in the `.map` but we'd need to pass in pretty much all the props (e.g., `item`, `attribute`, `refine`) to this component. That said, I'm not sure abstracting this out provides much value other than shortening the `FacetList` component, given its not reused anywhere else within the app.